### PR TITLE
add confidence interval

### PR DESCRIPTION
**Pull request by hnyu**
Add code for plotting confidence intervals. Also add average_y as a property of MeanCurve.

### CODE REVIEW

**Comment by emailweixu** [comment]
> Add code for plotting confidence intervals. Also add average_y as a property of MeanCurve.

Do you mean remove average_y?

**Comment by hnyu** [comment]
> > Add code for plotting confidence intervals. Also add average_y as a property of MeanCurve.
> 
> Do you mean remove average_y?

The function average_y() was removed. But now MeanCurve inherently has ay, min_ay, max_ay as properties when being constructed.

File: alf/utils/plot_tb_curves.py [hunk 1]
```diff
@@ -287,40 +336,49 @@ class MeanCurveGroupReader(object):
...
         Args:
             mean_curve_readers (list[MeanCurveReader]): a list of
-                ``MeanCurveReader`` of multiple tasks for one method.
+                ``MeanCurveReader`` of multiple tasks for one method. It's the
+                user's responsibility to ensure that it's meaningful to
+                group these task event files and show their mean and variance.
             task_performance_ranges (list[tuple(float)]): a list of tuples, where
                 each tuple is a pair of floats used for normalizing the corresponding
-                task.
+                task. If None, each tuple will be set to ``(0,1)`` (no effect).
```
Comment (emailweixu) [inline alf/utils/plot_tb_curves.py]: "If None, no normalization will be performed."

File: alf/utils/plot_tb_curves.py [hunk 2]
```diff
@@ -27,23 +27,83 @@
...
+            x=x,
+            y=y,
+            min_y=min_y,
+            max_y=max_y,
+            ay=ay,
+            min_ay=min_ay,
+            max_ay=max_ay,
+            name=name)
 
     def final_y(self, N=1):
```
Comment (emailweixu) [inline alf/utils/plot_tb_curves.py:104]: Seems that this function is not used anywhere.

Comment (hnyu) [reply alf/utils/plot_tb_curves.py:104]: > Seems that this function is not used anywhere.

I used this in my plotting scripts.

**Review by emailweixu — APPROVED** [review]

**Pull request merged**